ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -147,8 +161,24 @@ pub(crate) struct Instruction {
     name: RefCell<Option<String>>,
 }
 
+impl Instruction {
+    fn get_type<'a>(&self, m: &'a AOTModule) -> &'a Type {
+        &m.types[self.type_index]
```

Comment:
There's a clear dependency between `&self` and `&AOTModule`: if you `get_type` on the wrong module, this will do weird things. That should, at least, be documented IMHO. Personally, I would probably put `type(&self, instr: Instruction)` on `AOTModule` because it then becomes more obvious that you really need to think carefully about which `Instruction`s came from which `AOTModule`.

Reply:
[ccc5797](https://github.com/ykjit/yk/pull/853/commits/ccc5797d7a2c66f524a1a367f1f33b0cdbdb7e41) does as you suggested,  then [bbe01b8](https://github.com/ykjit/yk/pull/853/commits/bbe01b88694d09d5e1f9f4382d24088b4195c986) does the same to another part of the IR.

I now wonder if we should have a `Typed` trait so that the AOT module doesn't need one method per kind of typed thing that you might want to get the type of. Then again, we've only needed to ask the type of those two things so far...

Thoughts? Shall we kick it down the road a bit?

Reply:
Yeah, probably worth punting for now.